### PR TITLE
Feature/upload with arseeding

### PR DESCRIPTION
**Pull request by xudean**

### CODE REVIEW

File: src/demo/data_provider_arseeding.ts
Comment (fksyuan) [inline src/demo/data_provider_arseeding.ts:1]: The file must merget to data_provider.ts

File: src/index.ts [hunk 1]
```diff
@@ -18,14 +24,16 @@ import { transferAOCREDToTask } from './processes/utils';
  * @param priceInfo - The data price symbol(symbol is optional, default is AOCRED) and price. Currently only AO's test token AOCRED is supported, with a minimum price unit of 1 (1 means 0.001 AOCRED)
  * @param wallet - The ar wallet json object, this wallet must have AR Token. Pass `window.arweaveWallet` in a browser
  * @param arweave - Arweave object generated by arweave-js init method and default is AR production
+ * @param extParam
```
Comment (fksyuan) [inline src/index.ts]: Should explain AR and arseeding params

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,4 @@
 .DS_Store
 /.vscode
 *-key.json
+./package-lock.json
```
Comment (fksyuan) [inline .gitignore:6]: Delete package-lock.json in codebase

**Review by fksyuan — CHANGES_REQUESTED** [review]

File: src/demo/data_provider_arseeding.ts
Comment (xudean) [reply src/demo/data_provider_arseeding.ts:1]: Using a separate document has better conditioning and helps users understand how to use

File: src/index.ts (hunk 1 above)
Comment (xudean) [reply src/index.ts]: fixed

File: .gitignore (hunk 2 above)
Comment (xudean) [reply .gitignore:6]: fixed

**Review by fksyuan — APPROVED** [review]

**Pull request merged**